packages/utilities-react/src/StepFlow/StepContext.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useContext, createContext, ReactNode, useState } from 'react';
+import { StepContextType } from './types';
+
+const StepContext = createContext<StepContextType | undefined>(undefined);
+
+interface StepProviderProps {
+  children?: ReactNode;
+}
+
+export const StepProvider = ({ children }: StepProviderProps) => {
+  const [totalSteps, setTotalSteps] = useState(1);
+  const [currentStep, setCurrentStep] = useState(1);
+  const [formState, setFormState] = useState({});
+
+  const context: StepContextType = {
+    formState,
+    setFormState,
+    totalSteps,
+    setTotalSteps,
+    currentStep,
+    handleGoToStep: (step) => setCurrentStep(step),
+    handleNext: () => setCurrentStep((step) => step + 1),
+    handlePrevious: () => setCurrentStep((step) => step - 1),
+  };
+
+  return (
+    <StepContext.Provider value={context}>{children}</StepContext.Provider>
+  );
+};
+
+export const useStepContext = (): StepContextType => {
+  const context = useContext<StepContextType | undefined>(StepContext);
```

Comment:
- Both types can be inferred.
- Is `StepContextType` necessary or can it be deleted to simplify the implementation? It's just renaming `StepState`.

```suggestion
export const useStepContext = () => {
  const context = useContext(StepContext);
```

Reply:
Addressed in https://github.com/carbon-design-system/carbon/pull/20389/commits/77aaf12e3020cc62c39c475d6cd8dc538f040f67